x/ccv/provider/keeper/grpc_query.go

```diff
@@ -49,26 +50,57 @@ func (k Keeper) QueryConsumerChains(goCtx context.Context, req *types.QueryConsu
 
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	chains := []*types.Chain{}
-	for _, chainID := range k.GetAllRegisteredConsumerIds(ctx) {
-		c, err := k.GetConsumerChain(ctx, chainID)
+	lastCID, ok := k.GetConsumerId(ctx)
```

Comment:
Maybe we can call `firstUnusedConsumerId` instead.

Reply:
Need more pointers to firstUnusedConsumerId. Can't find it.

Reply:
Meant that we can call `lastCID` to `firstUnusedConsumerId`.